rollastic/node.py

```diff
@@ -53,15 +56,26 @@ def populate(self):
         '''
         self.clear()
 
-        info = self.cluster.es.nodes.info(self.node_id)['nodes'].get(self.node_id, {})
-        if not info:
-            _LOG.warning('Bad result for node info. node_id=%s info=%s', self.node_id, info)
-        self.update(info)
+        try:
+            info = self.cluster.es.nodes.info(self.node_id)['nodes'].get(self.node_id, {})
+            self.update(info)
 
-        stats = self.cluster.es.nodes.stats(self.node_id)['nodes'].get(self.node_id, {})
-        if not stats:
-            _LOG.warning('Bad result for node stats. node_id=%s stats=%s', self.node_id, stats)
-        self.update(stats)
+            stats = self.cluster.es.nodes.stats(self.node_id)['nodes'].get(self.node_id, {})
+            self.update(stats)
+
+            if not stats or info:
+                raise Exception("Failed to retrieve values from node. node_id={0} info={1} stats={2}".format(
+                    self.node_id, info, stats))
+
+            # reset the counter to zero because we only care about tracking consecutive bad results from the node
+            self.bad_result_count = 0
+        except Exception as e:
+            self.bad_result_count += 1
+            _LOG.warning("{0} (failure {1} of {2}".format(str(e), self.bad_result_count, self.bad_result_maximum))
```

Comment:
Use sprintf that's already being handled by logging, otherwise it looks ugly (and is two different format string methods being applied)